errdefs/wrapped.go

```diff
@@ -0,0 +1,10 @@
+package errdefs
+
+// Causal is an error interface for errors which have wrapped another error
+// in a non-opaque way.
+//
+// This pattern is used by github.com/pkg/errors
+type Causal interface {
```

Comment:
Given pkg/errors is going to be standardized in go1.13, should we do our own, if it's going to be "obsoleted" in a few months? 
See: https://github.com/golang/go/issues/29934

Reply:
pkg/errors is a package used all over the place.
Having a solution in go (or really it's currently in golang.org/x/exp/errors) does not make the usage of pkg/errors go away.
Once the interface for x/exp/errors is finalized we should definitely support it here.

Reply:
I think the comment was misunderstood. Rather than defining our own causal interface, I was going to suggest using the causal interface from `pkg/errors`.

Reply:
Oh... yes... you mean use `errors.Cause`
I think I didn't want to take a dependency on pkg/errors in this package for such a small thing.

Reply:
I don't see anything wrong with introducing said dependency if it's something we're doing down the line.